unitycatalog/functions/databricks.py

```diff
@@ -0,0 +1,549 @@
+                It should follow the syntax of CREATE FUNCTION statement in Databricks.
+                Ref: https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-create-sql-function.html#syntax
+
+                .. Note:: This is supported using databricks connect; if databricks-connect version is 15.1.0 or above,
+                    by default it uses serverless compute; otherwise please provide a cluster_id when constructing the client
+                    and the client will use remote cluster to create the function.
+            function_info (CreateFunction, optional): The function info. Defaults to None.
+
+        Returns:
+            FunctionInfo: The created function info.
+        """
+        if sql_function_body and function_info:
+            raise ValueError("Only one of sql_function_body and function_info should be provided.")
+        if sql_function_body:
+            _logger.info("Using databricks connect to create function.")
+            self.set_default_spark_session()
+            try:
+                self.spark.sql(sql_function_body)  # type: ignore
+            except Exception as e:
+                raise ValueError(
+                    f"Failed to create function with sql body: {sql_function_body}"
+                ) from e
+            return self.retrieve_function(extract_function_name(sql_function_body))  # type: ignore
+        if function_info:
+            return self.client.functions.create(function_info)
+        raise ValueError("Either function_info or sql_function_body should be provided.")
+
+    @override
+    def retrieve_function(
+        self, function_name: str, **kwargs: Any
+    ) -> Union["FunctionInfo", List["FunctionInfo"]]:
+        splits = function_name.split(".")
+        if len(splits) != 3:
+            raise ValueError(
+                f"Invalid function name: {function_name}, expecting format <catalog_name>.<schema_name>.<function_name>."
+            )
+        if splits[-1] == "*":
+            functions = self.client.functions.list(catalog_name=splits[0], schema_name=splits[1])
+            return list(functions)
+        return self.client.functions.get(function_name, include_browse=kwargs.get("include_browse"))
+
+    @override
+    def _validate_param_type(self, value: Any, param_info: "FunctionParameterInfo") -> None:
+        value_python_type = column_type_to_python_type(param_info.type_name)
+        if not isinstance(value, value_python_type):
+            raise ValueError(
+                f"Parameter {param_info.name} should be of type {param_info.type_name.value} "
+                f"(corresponding python type {value_python_type}), but got {type(value)}"
+            )
+        validate_param(value, param_info)
+
+    @override
+    def execute_function(
+        self, function_name: str, parameters: Optional[Dict[str, Any]] = None, **kwargs: Any
+    ) -> FunctionExecutionResult:
```

Comment:
do we need to override this?

Reply:
I'm overriding it just for the docstring, is there any other ways to do it?

Reply:
got it, didn't notice that.